jujugui/static/gui/src/app/user/test-user.js

```diff
@@ -24,39 +24,54 @@ chai.config.truncateThreshold = 0;
 
 describe('user auth class', () => {
   it('exists', () => {
-    const user = new window.jujugui.User({storage: {}});
+    const user = new window.jujugui.User({sessionStorage: {}});
     assert.isObject(user);
   });
 
-  describe('controller credentials', () => {
-    const getMockStorage = function() {
-      return new function() {
-        return {
-          store: {},
-          setItem: function(name, val) { this.store[name] = val; },
-          getItem: function(name) { return this.store[name] || null; }
-        };
+  const getMockStorage = function() {
+    return new function() {
+      return {
+        store: {},
+        setItem: function(name, val) { this.store[name] = val; },
+        getItem: function(name) { return this.store[name] || null; }
       };
     };
+  };
+
+  describe('identity credentials', () => {
+    it('can be set', () => {
+      let storage = getMockStorage();
+      const user = new window.jujugui.User({localStorage: storage});
+      user.identity = 'doctor';
+      assert.equal(storage.store['discharge-token'], 'doctor');
+    });
+    it('can be retrieved', () => {
+      let storage = getMockStorage();
+      const user = new window.jujugui.User({localStorage: storage});
+      storage.store['discharge-token'] = 'doctor';
+      assert.equal(user.identity, 'doctor');
+    });
+  });
 
+  describe('controller credentials', () => {
     it('can be set', () => {
       let storage = getMockStorage();
```

Comment:
It seems that the first two lines of all tests are shared and can be moved to before method.